### PR TITLE
Build and test both C++ loggers where possible

### PR DESCRIPTION
**Pull request by wilhuff**
Add the means to build and test both log_stdio and log_apple where possible.

Also make log_apple build outside Xcode with a mild refactoring to make it smaller.

Note that this change is dependent on the build of FirebaseCore introduced in #594. Once that PR is submitted I'll retarget this PR at master.

### CODE REVIEW

File: Firestore/core/src/firebase/firestore/util/CMakeLists.txt [hunk 1]
```diff
@@ -22,3 +22,34 @@ add_library(
...
+    FirebaseCore
+  )
+
+  target_link_libraries(
+    firebase_firestore_util
+    PUBLIC
+    firebase_firestore_util_log_apple
+  )
+
+else(NOT APPLE)
```
Comment (zxu123) [inline Firestore/core/src/firebase/firestore/util/CMakeLists.txt]: Should add_library(firebase_firestore_util_log_stdio,...) be moved under else(NOT APPLE) here?

File: Firestore/core/test/firebase/firestore/util/log_test.cc [hunk 2]
```diff
@@ -22,6 +22,16 @@ namespace firebase {
 namespace firestore {
 namespace util {
 
+// When running against the log_apple.mm implementation (backed by FIRLogger)
+// this test can fail if debug_mode gets persisted in the user defaults. Check
+// for defaults getting in your way with
+//
+//   defaults read firebase_firestore_util_log_appleA_test
```
Comment (zxu123) [inline Firestore/core/test/firebase/firestore/util/log_test.cc]: typo? ..._appleA_...

**Comment by wilhuff** [comment]
I've addressed @zxu123's comments and merged master but this has triggered a review requirement again.

**Comment by zxu123** [comment]
LGTM

**Pull request merged**